snuba_sdk/mql/mql.py

```diff
@@ -32,8 +32,12 @@ class InvalidMQLQueryError(Exception):
 expression = term (_ expr_op _ term)*
 expr_op = "+" / "-"
 
-term = coefficient (_ term_op _ coefficient)*
+term = unary (_ term_op _ unary)*
 term_op = "*" / "/"
+
+unary = unary_op? coefficient
+unary_op = "-"
```

Comment:
I think the `number` rule should be changed to catch the negative sign. That will make negative scalars (`-1`) parse in a more logical way.

Reply:
In my implementation I support negation for numeric scalars and formulas/timeseries.

Reply:
That's true, but if you added changed the definition of a `number` to support negatives, you wouldn't need to manually negate the numbers in the `unary` code.

Also it would just conceptually make more sense: the `number` rule defines a number, instead of needing a the `number` and the `unary` rule in order to get a correct scalar value.

Reply:
Decided to keep the grammar as is since adding negation multiple times in the grammar is less clean.